jina/serve/networking.py

```diff
@@ -740,32 +810,41 @@ async def task_wrapper():
                         timeout=timeout,
                     )
                 except AioRpcError as e:
-                    self._handle_aiorpcerror(
-                        e=e,
+                    await self._handle_aiorpcerror(
+                        error=e,
                         retry_i=i,
                         request_id=requests[0].request_id,
-                        dest_addr=tried_addresses,
-                        num_retries=total_num_tries,
+                        tried_addresses=tried_addresses,
+                        total_num_tries=total_num_tries,
+                        current_address=current_connection.address,
+                        connection_list=connections,
                     )
 
         return asyncio.create_task(task_wrapper())
 
     def _send_discover_endpoint(
         self,
-        connection: ConnectionStubs,
+        connection_list: ReplicaList,
         timeout: Optional[float] = None,
     ) -> asyncio.Task:
         # this wraps the awaitable object from grpc as a coroutine so it can be used as a task
         # the grpc call function is not a coroutine but some _AioCall
         async def task_wrapper():
+            connection = None
+            if connection_list:
+                connection = await connection_list.get_next_connection()
```

Comment:
@JoanFM is there a particular reason this was previously done outside the `async` scope? I had to pull it into here, I think it should not be a problem, but maybe you want to check.